server/src/sas/CompletionProvider.ts

```diff
@@ -281,6 +284,381 @@ export class CompletionProvider {
+                if (activeParameter > curArugmentLength - 1) {
+                  syntaxArgumentArr.some((item, index) => {
+                    if (item && item.length - 1 > activeParameter) {
+                      activeSignature = index;
+                      return true;
+                    }
+                  });
+                }
+              }
+
+              const signatures: SignatureInformation[] = [];
+              syntaxArr.forEach((syntax, index) => {
+                const argumentsInSyntax = syntaxArgumentArr[index];
+                // match like the N in DIM <N> (array-name)
+                const outerArguments = syntax
+                  .match(new RegExp(`${keyword}\\s+(.*?)\\s+\\(`, "i"))?.[1]
+                  ?.replace(/</g, "")
+                  .replace(/>/g, "")
+                  .split(",")
+                  .map((argument) => argument.trim());
+                const outerArgumentsDocumentation = data.arguments
+                  ?.filter((item) => outerArguments?.includes(item.name))
+                  .map((item) => `**${item.name}:** ${item.description}`)
+                  .join("\n\n");
+                // the left arguments in data.arguments but ont in syntax
+                let leftArgumentsDocumentation = "";
+                if (
+                  argumentsInSyntax &&
+                  !argumentsInSyntax.every((item) => item[1] && item[2])
+                ) {
+                  const leftArguments = data.arguments
+                    ?.filter(
+                      (item) =>
+                        ![...allArguments, ...(outerArguments || [])].some(
+                          (_argument) =>
+                            _argument
+                              ?.replace("(s)", "")
+                              .replace(/['"‘’<>()|.]/g, "")
+                              .replace(/[-_\s]?\d$/, "")
+                              .replace(/[-_\s]n$/, "")
+                              .trim()
+                              .toLowerCase() ===
+                            item.name
+                              ?.replace("(s)", "")
+                              .replace(/['"‘’<>()|.]/g, "")
+                              .replace(/[-_\s]?\d$/, "")
+                              .replace(/[-_\s]n$/, "")
+                              .trim()
+                              .toLowerCase(),
+                        ) && item.description,
+                    )
+                    .map((item) => `**${item.name}:** ${item.description}`);
+                  if (leftArguments) {
+                    leftArgumentsDocumentation = [
+                      ...new Set(leftArguments),
+                    ].join("\n\n");
+                  }
+                }
+
+                signatures.push({
+                  label: syntax,
+                  documentation: data.data,
+                  parameters: argumentsInSyntax?.map((item) => {
```

Comment:
deconstruct the `item` to be self explained, for example `([name, description, isMatch]) => {`